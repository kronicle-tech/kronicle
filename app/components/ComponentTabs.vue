<template>
  <div class="my-3">
    <b-button class="my-1" :to="`/components/${componentId}`"> Info </b-button>
    <b-button class="my-1" :to="`/components/${componentId}/diagrams`">
      Diagrams
    </b-button>
    <b-button
      v-if="componentAvailableData.hasNodes"
      class="my-1"
      :to="`/components/${componentId}/response-times`"
    >
      Response Times
    </b-button>
    <b-button
      v-if="componentAvailableData.hasNodes"
      class="my-1"
      :to="`/components/${componentId}/downstream-response-times`"
    >
      Downstream Response Times
    </b-button>
    <b-button
      v-if="componentAvailableData.hasCallGraphs"
      class="my-1"
      :to="`/components/${componentId}/call-graphs`"
    >
      Call Graphs
    </b-button>
    <b-button
      v-if="hasMetadataType('cross-functional-requirement')"
      class="my-1"
      :to="`/components/${componentId}/cross-functional-requirements`"
    >
      Cross Functional Requirements
    </b-button>
    <b-button
      v-if="hasMetadataType('tech-debt')"
      class="my-1"
      :to="`/components/${componentId}/tech-debts`"
    >
      Tech Debts
    </b-button>
    <b-button
      v-if="hasMetadataType('scanner-error')"
      class="my-1"
      :to="`/components/${componentId}/errors`"
    >
      Errors
    </b-button>
    <b-button
      v-if="hasStateType('openapi-specs')"
      class="my-1"
      :to="`/components/${componentId}/openapi-specs`"
    >
      OpenAPI
    </b-button>
    <b-button
      v-if="hasStateType('graphql-schemas')"
      class="my-1"
      :to="`/components/${componentId}/graphql-schemas`"
    >
      GraphQL
    </b-button>
    <b-button
      v-if="hasStateType('git-repo')"
      class="my-1"
      :to="`/components/${componentId}/repo`"
    >
      Repo
    </b-button>
    <b-button
      v-if="hasStateType('readme')"
      class="my-1"
      :to="`/components/${componentId}/readme`"
    >
      README
    </b-button>
    <b-button
      v-if="hasStateType('to-dos')"
      class="my-1"
      :to="`/components/${componentId}/to-dos`"
    >
      To Dos
    </b-button>
    <b-button
      v-if="hasStateType('lines-of-code')"
      class="my-1"
      :to="`/components/${componentId}/lines-of-code`"
    >
      Lines of Code
    </b-button>
    <b-button
      v-if="hasStateType('software-repositories')"
      class="my-1"
      :to="`/components/${componentId}/software-repositories`"
    >
      Software Repositories
    </b-button>
    <b-button
      v-if="hasStateType('softwares')"
      class="my-1"
      :to="`/components/${componentId}/software`"
    >
      Software
    </b-button>
    <b-button
      v-if="hasStateType('imports')"
      class="my-1"
      :to="`/components/${componentId}/imports`"
    >
      Imports
    </b-button>
    <b-button
      v-if="hasMetadataType('test-result')"
      class="my-1"
      :to="`/components/${componentId}/tests`"
    >
      Tests
    </b-button>
  </div>
</template>

<script lang="ts">
import Vue, { PropType } from 'vue'
import { BButton } from 'bootstrap-vue'
import { ComponentAvailableData } from '~/src/fetchComponentAvailableData'

export default Vue.extend({
  components: {
    'b-button': BButton,
  },
  props: {
    componentId: {
      type: String,
      required: true,
    },
    componentAvailableData: {
      type: Object as PropType<ComponentAvailableData>,
      required: true,
    },
  },
  methods: {
    hasMetadataType(metadataType: string): boolean {
      return this.componentAvailableData.metadataTypes.includes(metadataType)
    },
    hasStateType(stateType: string): boolean {
      return this.componentAvailableData.stateTypes.includes(stateType)
    },
  },
})
</script>
