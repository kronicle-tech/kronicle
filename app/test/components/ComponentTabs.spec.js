import { mount } from '@vue/test-utils'
import ComponentTabs from '@/components/ComponentTabs.vue'

describe('ComponentTabs', () => {
  let propsData
  let wrapper
  const createWrapper = () => {
    wrapper = mount(ComponentTabs, { propsData })
  }

  beforeEach(() => {
    propsData = {
      componentId: 'test-component-id-1',
      stateTypes: [],
    }
  })

  afterEach(() => {
    wrapper.destroy()
    wrapper = null
  })

  describe('when there are no state types', () => {
    beforeEach(() => {
      propsData.stateTypes = []
    })

    test('renders the component with less tabs', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when there are no state types', () => {
    beforeEach(() => {
      propsData.stateTypes = [
        'lines-of-code',
      ]
    })

    test('renders the component with all tabs', () => {
      createWrapper()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
