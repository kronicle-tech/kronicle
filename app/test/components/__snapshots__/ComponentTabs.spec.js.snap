// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentTabs renders the component 1`] = `
"<div class=\\"my-3\\"><a href=\\"/components/test-component-id-1\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">Info</a> <a href=\\"/components/test-component-id-1/dependencies\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Dependencies
  </a> <a href=\\"/components/test-component-id-1/response-times\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Response Times
  </a> <a href=\\"/components/test-component-id-1/downstream-response-times\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Downstream Response Times
  </a> <a href=\\"/components/test-component-id-1/call-graphs\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Call Graphs
  </a> <a href=\\"/components/test-component-id-1/cross-functional-requirements\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Cross Functional Requirements
  </a> <a href=\\"/components/test-component-id-1/tech-debts\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Tech Debts
  </a> <a href=\\"/components/test-component-id-1/tests\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">Tests</a> <a href=\\"/components/test-component-id-1/errors\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">Errors</a> <a href=\\"/components/test-component-id-1/openapi-specs\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    OpenAPI Specs
  </a> <a href=\\"/components/test-component-id-1/repo\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">Repo</a> <a href=\\"/components/test-component-id-1/readme\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">README</a> <a href=\\"/components/test-component-id-1/to-dos\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">To Dos</a> <a href=\\"/components/test-component-id-1/lines-of-code\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Lines of Code
  </a> <a href=\\"/components/test-component-id-1/software-repositories\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">
    Software Repositories
  </a> <a href=\\"/components/test-component-id-1/software\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">Software</a> <a href=\\"/components/test-component-id-1/imports\\" target=\\"_self\\" class=\\"btn my-1 btn-secondary\\">Imports</a></div>"
`;
